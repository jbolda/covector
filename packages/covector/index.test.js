const { covector } = require("./index");
const { main } = require("effection");
const mockConsole = require("jest-mock-console");
const fixtures = require("fixturez");
const f = fixtures(__dirname);

describe("integration test", () => {
  it("passes correct config for js and rust", async () => {
    const restoreConsole = mockConsole(["log", "dir"]);
    const fullIntegration = f.copy("integration.js-and-rust-with-changes");
    const covectored = await main(
      covector({
        command: "status",
        cwd: fullIntegration,
      })
    );
    expect({
      consoleLog: console.log.mock.calls,
      consoleDir: console.dir.mock.calls,
      covectorReturn: covectored,
    }).toMatchSnapshot();
    restoreConsole();
  });

  it("runs version for js and rust", async () => {
    const restoreConsole = mockConsole(["log", "info"]);
    const fullIntegration = f.copy("integration.js-and-rust-with-changes");
    const covectored = await main(
      covector({
        command: "version",
        cwd: fullIntegration,
      })
    );
    expect({
      consoleLog: console.log.mock.calls,
      consoleInfo: console.info.mock.calls,
      covectorReturn: covectored.map((pkg) => {
        // remove these as they are dependent on the OS
        // and user running them so would always fail
        delete pkg.vfile.history;
        delete pkg.vfile.cwd;
        return pkg;
      }),
    }).toMatchSnapshot();
    restoreConsole();
  });

  it("runs publish for js and rust", async () => {
    const restoreConsole = mockConsole(["log", "info"]);
    const fullIntegration = f.copy("integration.js-and-rust-with-changes");
    const covectored = await main(
      covector({
        command: "publish",
        cwd: fullIntegration,
      })
    );
    expect({
      consoleLog: console.log.mock.calls,
      consoleInfo: console.info.mock.calls,
      covectorReturn: covectored,
    }).toMatchSnapshot();
    restoreConsole();
  });

  it("fails with error", async () => {
    const restoreConsole = mockConsole(["log", "info"]);
    const fullIntegration = f.copy("integration.js-with-publish-error");
    const covectored = await main(
      covector({
        command: "publish",
        cwd: fullIntegration,
      })
    );
    expect({
      consoleLog: console.log.mock.calls,
      consoleInfo: console.info.mock.calls,
      covectorReturn: covectored,
    }).toMatchSnapshot();
    expect(covectored).toThrow();
    restoreConsole();
  });
});
