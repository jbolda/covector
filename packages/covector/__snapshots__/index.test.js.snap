// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test fails with error 1`] = `
Object {
  "consoleInfo": Array [],
  "consoleLog": Array [
    Array [
      "publishing tauri.js with npm publish",
    ],
  ],
  "covectorReturn": Object {
    "tauri.js": true,
  },
}
`;

exports[`integration test passes correct config for js and rust 1`] = `
Object {
  "consoleDir": Array [
    Array [
      Array [
        Object {
          "releases": Object {
            "tauri": "minor",
          },
          "summary": "Summary about the changes in tauri",
        },
      ],
    ],
    Array [
      Array [
        Object {
          "releases": Object {
            "tauri-updater": "patch",
          },
          "summary": "Summary about the changes in tauri-updater",
        },
      ],
    ],
  ],
  "consoleLog": Array [
    Array [
      "changes:",
    ],
    Array [
      "tauri => minor",
    ],
    Array [
      "tauri-updater => patch",
    ],
  ],
  "covectorReturn": "There are 2 changes which include tauri with minor, tauri-updater with patch",
}
`;

exports[`integration test runs publish for js and rust 1`] = `
Object {
  "consoleInfo": Array [],
  "consoleLog": Array [
    Array [
      "publishing tauri.js with echo publishing tauri.js would happen here",
    ],
    Array [
      "publishing tauri-bundler with echo publishing tauri-bundler would happen here",
    ],
    Array [
      "publishing tauri with echo publishing tauri would happen here",
    ],
    Array [
      "publishing tauri-api with echo publishing tauri-api would happen here",
    ],
    Array [
      "publishing tauri-utils with echo publishing tauri-utils would happen here",
    ],
  ],
  "covectorReturn": Object {
    "tauri": true,
    "tauri-api": true,
    "tauri-bundler": true,
    "tauri-updater": false,
    "tauri-utils": true,
    "tauri.js": true,
  },
}
`;

exports[`integration test runs version for js and rust 1`] = `
Object {
  "consoleInfo": Array [
    Array [
      ".changes/first-change.md was deleted",
    ],
    Array [
      ".changes/second-change.md was deleted",
    ],
  ],
  "consoleLog": Array [
    Array [
      "bumping tauri with minor",
    ],
    Array [
      "bumping tauri-updater with patch",
    ],
    Array [
      "bumping tauri.js with minor",
    ],
    Array [
      "Could not load the CHANGELOG.md. Creating one.",
    ],
    Array [
      "Could not load the CHANGELOG.md. Creating one.",
    ],
    Array [
      "Could not load the CHANGELOG.md. Creating one.",
    ],
  ],
  "covectorReturn": Array [
    Object {
      "name": "tauri",
      "pkg": Object {
        "build-dependencies": Object {
          "tauri_includedir_codegen": "0.5.2",
        },
        "dependencies": Object {
          "anyhow": "1.0.31",
          "base64": "0.12.1",
          "envmnt": "0.8.2",
          "lazy_static": "1.4.0",
          "phf": "0.8.0",
          "serde": Object {
            "__tomlInline": true,
            "features": Array [
              "derive",
            ],
            "version": "1.0",
          },
          "serde_json": "1.0",
          "tauri-api": Object {
            "__tomlInline": true,
            "path": "../tauri-api",
            "version": "0.5",
          },
          "tauri_includedir": "0.5.0",
          "thiserror": "1.0.19",
          "threadpool": "1.8",
          "tiny_http": Object {
            "__tomlInline": true,
            "optional": true,
            "version": "0.7",
          },
          "uuid": Object {
            "__tomlInline": true,
            "features": Array [
              "v4",
            ],
            "version": "0.8.1",
          },
          "web-view": "=0.6.2",
          "webbrowser": "0.5.2",
          "webview-sys": "=0.5.0",
        },
        "dev-dependencies": Object {
          "proptest": "0.9.6",
          "serde": Object {
            "__tomlInline": true,
            "features": Array [
              "derive",
            ],
            "version": "1.0",
          },
          "serde_json": "1.0",
          "tauri": Object {
            "__tomlInline": true,
            "features": Array [
              "all-api",
              "edge",
            ],
            "path": ".",
          },
        },
        "example": Array [
          Object {
            "name": "communication",
            "path": "examples/communication/src-tauri/src/main.rs",
          },
        ],
        "features": Object {
          "all-api": Array [],
          "copy-file": Array [],
          "create-dir": Array [],
          "edge": Array [
            "web-view/edge",
          ],
          "embedded-server": Array [
            "tiny_http",
          ],
          "event": Array [],
          "execute": Array [],
          "no-server": Array [],
          "open": Array [],
          "open-dialog": Array [],
          "read-binary-file": Array [],
          "read-dir": Array [],
          "read-text-file": Array [],
          "remove-dir": Array [],
          "remove-file": Array [],
          "rename-file": Array [],
          "save-dialog": Array [],
          "set-title": Array [],
          "updater": Array [],
          "write-file": Array [],
        },
        "package": Object {
          "authors": "test",
          "description": "Make tiny, secure apps for all desktop platforms with Tauri",
          "edition": "2018",
          "exclude": Array [
            "test/fixture/**",
          ],
          "homepage": "https://tauri.studio",
          "license": "MIT",
          "metadata": Object {
            "docs": Object {
              "rs": Object {
                "features": Array [
                  "all-api",
                ],
              },
            },
          },
          "name": "tauri",
          "repository": "https://github.com/tauri-apps/tauri",
          "version": "0.6.0",
        },
      },
      "version": "0.6.0",
      "vfile": VFile {
        "contents": "[package]
name = \\"tauri\\"
version = \\"0.5.2\\"
authors = \\"test\\"
license = \\"MIT\\"
homepage = \\"https://tauri.studio\\"
repository = \\"https://github.com/tauri-apps/tauri\\"
description = \\"Make tiny, secure apps for all desktop platforms with Tauri\\"
edition = \\"2018\\"
exclude = [\\"test/fixture/**\\"]

[dependencies]
serde_json = \\"1.0\\"
serde = { version = \\"1.0\\", features = [\\"derive\\"] }
webview-sys = \\"=0.5.0\\"
web-view = \\"=0.6.2\\"
tauri_includedir = \\"0.5.0\\"
phf = \\"0.8.0\\"
base64 = \\"0.12.1\\"
webbrowser = \\"0.5.2\\"
lazy_static = \\"1.4.0\\"
tiny_http = { version = \\"0.7\\", optional = true }
threadpool = \\"1.8\\"
uuid = { version = \\"0.8.1\\", features = [\\"v4\\"] }
anyhow = \\"1.0.31\\"
thiserror = \\"1.0.19\\"
envmnt = \\"0.8.2\\"

tauri-api = { version = \\"0.5\\",  path = \\"../tauri-api\\" }

[build-dependencies]
tauri_includedir_codegen = \\"0.5.2\\"

[dev-dependencies]
proptest = \\"0.9.6\\"
serde_json = \\"1.0\\"
tauri = {path = \\".\\", features = [ \\"all-api\\", \\"edge\\" ]}
serde = { version = \\"1.0\\", features = [ \\"derive\\" ] }

[features]
edge = [\\"web-view/edge\\"]
embedded-server = [\\"tiny_http\\"]
no-server = []
all-api = []
read-text-file = []
read-binary-file = []
write-file = []
read-dir = []
copy-file = []
create-dir = []
remove-dir = []
remove-file = []
rename-file = []
set-title = []
execute = []
open = []
event = []
updater = []
open-dialog = []
save-dialog = []

[package.metadata.docs.rs]
features = [\\"all-api\\"]

[[example]]
name = \\"communication\\"
path = \\"examples/communication/src-tauri/src/main.rs\\"",
        "data": Object {},
        "messages": Array [],
      },
    },
    Object {
      "name": "tauri",
      "pkg": Object {
        "bin": Object {
          "tauri": "./bin/tauri.js",
        },
        "bugs": Object {
          "url": "https://github.com/tauri-apps/tauri/issues",
        },
        "contributors": Array [
          "test",
        ],
        "dependencies": Object {
          "@tauri-apps/tauri-inliner": "1.14.1",
          "@tauri-apps/toml": "2.2.4",
          "chalk": "4.0.0",
          "chokidar": "3.4.0",
          "cross-spawn": "7.0.3",
          "fast-glob": "3.2.2",
          "fs-extra": "9.0.1",
          "http-proxy": "1.18.1",
          "imagemin": "7.0.1",
          "imagemin-optipng": "8.0.0",
          "imagemin-pngquant": "9.0.0",
          "imagemin-zopfli": "7.0.0",
          "is-png": "2.0.0",
          "isbinaryfile": "4.0.6",
          "jsdom": "16.2.2",
          "lodash": "4.17.15",
          "minimist": "1.2.5",
          "ms": "2.1.2",
          "png2icons": "2.0.1",
          "read-chunk": "3.2.0",
          "sharp": "0.25.3",
          "webpack-merge": "4.2.2",
          "webpack-shell-plugin": "0.5.0",
        },
        "description": "Multi-binding collection of libraries and templates for building Tauri apps",
        "devDependencies": Object {
          "@babel/core": "7.10.2",
          "@babel/preset-env": "7.10.2",
          "@babel/preset-typescript": "7.10.1",
          "@types/cross-spawn": "6.0.2",
          "@types/fs-extra": "9.0.1",
          "@types/http-proxy": "1.17.4",
          "@types/imagemin": "7.0.0",
          "@types/imagemin-optipng": "5.2.0",
          "@types/jsdom": "16.2.3",
          "@types/lodash": "4.14.155",
          "@types/ms": "0.7.31",
          "@types/sharp": "0.25.0",
          "@types/webpack-merge": "4.1.5",
          "@typescript-eslint/eslint-plugin": "3.1.0",
          "@typescript-eslint/parser": "3.1.0",
          "babel-jest": "26.0.1",
          "dotenv": "8.2.0",
          "eslint": "7.1.0",
          "eslint-config-standard-with-typescript": "18.0.2",
          "eslint-plugin-import": "2.20.2",
          "eslint-plugin-lodash-template": "0.18.0",
          "eslint-plugin-node": "11.1.0",
          "eslint-plugin-promise": "4.2.1",
          "eslint-plugin-security": "1.4.0",
          "eslint-plugin-standard": "4.0.1",
          "husky": "4.2.5",
          "is-running": "2.1.0",
          "jest": "26.0.1",
          "jest-mock-process": "1.4.0",
          "lint-staged": "10.2.8",
          "lockfile-lint": "4.3.6",
          "promise": "8.1.0",
          "raw-loader": "4.0.1",
          "ts-loader": "7.0.5",
          "typescript": "3.9.3",
          "webpack": "4.43.0",
          "webpack-cli": "3.3.11",
          "webpack-node-externals": "1.7.2",
        },
        "funding": Object {
          "type": "opencollective",
          "url": "https://opencollective.com/tauri",
        },
        "homepage": "https://github.com/tauri-apps/tauri#readme",
        "husky": Object {
          "hooks": Object {
            "pre-commit": "lint-staged",
          },
        },
        "license": "MIT",
        "lint-staged": Array [
          "eslint --fix",
        ],
        "name": "tauri",
        "repository": Object {
          "type": "git",
          "url": "git+https://github.com/tauri-apps/tauri.git",
        },
        "scripts": Object {
          "build": "webpack --progress",
          "build-release": "yarn build --display none --progress false",
          "build:tauri[rust]": "cd ../tauri && TAURI_DIST_DIR=../../test/fixture/dist TAURI_DIR=../test/fixture cargo publish --dry-run --allow-dirty",
          "lint": "eslint --ext ts ./src/**/*.ts",
          "lint-fix": "eslint --fix --ext ts ./src/**/*.ts",
          "lint:lockfile": "lockfile-lint --path yarn.lock --type yarn --validate-https --allowed-hosts npm yarn",
          "prepublishOnly": "yarn build-release",
          "pretest": "yarn build",
          "test": "jest --runInBand --no-cache --testPathIgnorePatterns=\\"(build|dev)\\"",
          "test:local": "jest --runInBand",
        },
        "version": "0.7.0",
      },
      "version": "0.7.0",
      "vfile": VFile {
        "contents": "{
  \\"name\\": \\"tauri\\",
  \\"version\\": \\"0.6.2\\",
  \\"description\\": \\"Multi-binding collection of libraries and templates for building Tauri apps\\",
  \\"bin\\": {
    \\"tauri\\": \\"./bin/tauri.js\\"
  },
  \\"funding\\": {
    \\"type\\": \\"opencollective\\",
    \\"url\\": \\"https://opencollective.com/tauri\\"
  },
  \\"scripts\\": {
    \\"build\\": \\"webpack --progress\\",
    \\"build-release\\": \\"yarn build --display none --progress false\\",
    \\"test\\": \\"jest --runInBand --no-cache --testPathIgnorePatterns=\\\\\\"(build|dev)\\\\\\"\\",
    \\"pretest\\": \\"yarn build\\",
    \\"prepublishOnly\\": \\"yarn build-release\\",
    \\"test:local\\": \\"jest --runInBand\\",
    \\"lint\\": \\"eslint --ext ts ./src/**/*.ts\\",
    \\"lint-fix\\": \\"eslint --fix --ext ts ./src/**/*.ts\\",
    \\"lint:lockfile\\": \\"lockfile-lint --path yarn.lock --type yarn --validate-https --allowed-hosts npm yarn\\",
    \\"build:tauri[rust]\\": \\"cd ../tauri && TAURI_DIST_DIR=../../test/fixture/dist TAURI_DIR=../test/fixture cargo publish --dry-run --allow-dirty\\"
  },
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+https://github.com/tauri-apps/tauri.git\\"
  },
  \\"contributors\\": [
    \\"test\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/tauri-apps/tauri/issues\\"
  },
  \\"homepage\\": \\"https://github.com/tauri-apps/tauri#readme\\",
  \\"dependencies\\": {
    \\"@tauri-apps/tauri-inliner\\": \\"1.14.1\\",
    \\"@tauri-apps/toml\\": \\"2.2.4\\",
    \\"chalk\\": \\"4.0.0\\",
    \\"chokidar\\": \\"3.4.0\\",
    \\"cross-spawn\\": \\"7.0.3\\",
    \\"fast-glob\\": \\"3.2.2\\",
    \\"fs-extra\\": \\"9.0.1\\",
    \\"http-proxy\\": \\"1.18.1\\",
    \\"imagemin\\": \\"7.0.1\\",
    \\"imagemin-optipng\\": \\"8.0.0\\",
    \\"imagemin-pngquant\\": \\"9.0.0\\",
    \\"imagemin-zopfli\\": \\"7.0.0\\",
    \\"is-png\\": \\"2.0.0\\",
    \\"isbinaryfile\\": \\"4.0.6\\",
    \\"jsdom\\": \\"16.2.2\\",
    \\"lodash\\": \\"4.17.15\\",
    \\"minimist\\": \\"1.2.5\\",
    \\"ms\\": \\"2.1.2\\",
    \\"png2icons\\": \\"2.0.1\\",
    \\"read-chunk\\": \\"3.2.0\\",
    \\"sharp\\": \\"0.25.3\\",
    \\"webpack-merge\\": \\"4.2.2\\",
    \\"webpack-shell-plugin\\": \\"0.5.0\\"
  },
  \\"devDependencies\\": {
    \\"@babel/core\\": \\"7.10.2\\",
    \\"@babel/preset-env\\": \\"7.10.2\\",
    \\"@babel/preset-typescript\\": \\"7.10.1\\",
    \\"@types/cross-spawn\\": \\"6.0.2\\",
    \\"@types/fs-extra\\": \\"9.0.1\\",
    \\"@types/http-proxy\\": \\"1.17.4\\",
    \\"@types/imagemin\\": \\"7.0.0\\",
    \\"@types/imagemin-optipng\\": \\"5.2.0\\",
    \\"@types/jsdom\\": \\"16.2.3\\",
    \\"@types/lodash\\": \\"4.14.155\\",
    \\"@types/ms\\": \\"0.7.31\\",
    \\"@types/sharp\\": \\"0.25.0\\",
    \\"@types/webpack-merge\\": \\"4.1.5\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"3.1.0\\",
    \\"@typescript-eslint/parser\\": \\"3.1.0\\",
    \\"babel-jest\\": \\"26.0.1\\",
    \\"dotenv\\": \\"8.2.0\\",
    \\"eslint\\": \\"7.1.0\\",
    \\"eslint-config-standard-with-typescript\\": \\"18.0.2\\",
    \\"eslint-plugin-import\\": \\"2.20.2\\",
    \\"eslint-plugin-lodash-template\\": \\"0.18.0\\",
    \\"eslint-plugin-node\\": \\"11.1.0\\",
    \\"eslint-plugin-promise\\": \\"4.2.1\\",
    \\"eslint-plugin-security\\": \\"1.4.0\\",
    \\"eslint-plugin-standard\\": \\"4.0.1\\",
    \\"husky\\": \\"4.2.5\\",
    \\"is-running\\": \\"2.1.0\\",
    \\"jest\\": \\"26.0.1\\",
    \\"jest-mock-process\\": \\"1.4.0\\",
    \\"lint-staged\\": \\"10.2.8\\",
    \\"lockfile-lint\\": \\"4.3.6\\",
    \\"promise\\": \\"8.1.0\\",
    \\"raw-loader\\": \\"4.0.1\\",
    \\"ts-loader\\": \\"7.0.5\\",
    \\"typescript\\": \\"3.9.3\\",
    \\"webpack\\": \\"4.43.0\\",
    \\"webpack-cli\\": \\"3.3.11\\",
    \\"webpack-node-externals\\": \\"1.7.2\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\"
    }
  },
  \\"lint-staged\\": [
    \\"eslint --fix\\"
  ]
}
",
        "data": Object {},
        "messages": Array [],
      },
    },
    Object {
      "name": "tauri-updater",
      "pkg": Object {
        "dependencies": Object {
          "anyhow": "1.0.31",
          "attohttpc": Object {
            "__tomlInline": true,
            "features": Array [
              "json",
              "compress",
            ],
            "version": "0.10.1",
          },
          "serde": "1.0",
          "serde_json": "1.0",
          "tauri-api": Object {
            "__tomlInline": true,
            "path": "../tauri-api",
            "version": "0.5",
          },
          "tauri-utils": Object {
            "__tomlInline": true,
            "path": "../tauri-utils",
            "version": "0.5",
          },
          "tempdir": "0.3",
          "thiserror": "1.0.19",
          "zip": "0.5.3",
        },
        "package": Object {
          "authors": Array [
            "Lucas Fernandes Gonçalves Nogueira <lucas@tauri.studio>",
            "Daniel Thompson-Yvetot <denjell@sfosc.org>",
            "Tensor Programming <tensordeveloper@gmail.com>",
          ],
          "description": "Updater for Tauri",
          "edition": "2018",
          "exclude": Array [
            "test/fixture/**",
          ],
          "homepage": "https://tauri.studio",
          "license": "MIT",
          "name": "tauri-updater",
          "repository": "https://github.com/tauri-apps/tauri",
          "version": "0.4.3",
        },
      },
      "version": "0.4.3",
      "vfile": VFile {
        "contents": "[package]
name = \\"tauri-updater\\"
version = \\"0.4.2\\"
authors = [\\"Lucas Fernandes Gonçalves Nogueira <lucas@tauri.studio>\\", \\"Daniel Thompson-Yvetot <denjell@sfosc.org>\\", \\"Tensor Programming <tensordeveloper@gmail.com>\\"]
license = \\"MIT\\"
homepage = \\"https://tauri.studio\\"
repository = \\"https://github.com/tauri-apps/tauri\\"
description = \\"Updater for Tauri\\"
edition = \\"2018\\"
exclude = [\\"test/fixture/**\\"]

[dependencies]
attohttpc = {version = \\"0.10.1\\", features=[\\"json\\", \\"compress\\" ]}
# pbr = \\"1\\"
serde_json = \\"1.0\\"
serde = \\"1.0\\"
zip = \\"0.5.3\\"
tempdir = \\"0.3\\"
tauri-api = { version = \\"0.5\\", path = \\"../tauri-api\\" }
tauri-utils = { version = \\"0.5\\", path = \\"../tauri-utils\\" }
anyhow = \\"1.0.31\\"
thiserror = \\"1.0.19\\"
",
        "data": Object {},
        "messages": Array [],
      },
    },
  ],
}
`;
